CodeAnalyzerTool/CodeAnalyzerService.UI/code-analyzer-service.ui/src/pages/RuleViolationDetails.tsx

```diff
@@ -0,0 +1,92 @@
+import {useParams} from "react-router-dom";
+import {useRuleViolation} from "../hooks/useRuleViolation";
+import Loading from "../components/Loading";
+import {Alert, Box, Container, Divider, Grid, Paper, Stack, Typography, useTheme} from "@mui/material";
+import React, {ReactElement} from "react";
+import CategoryIcon from '@mui/icons-material/Category';
+import RuleNameIcon from '@mui/icons-material/Fingerprint';
+import ErrorIcon from '@mui/icons-material/Error';
+import WarningIcon from '@mui/icons-material/Warning';
+import InfoIcon from '@mui/icons-material/Info';
+import PluginIcon from '@mui/icons-material/Extension';
+import LanguageIcon from '@mui/icons-material/Terminal';
+import {SEVERITY} from "../model/Severity";
+import {RuleViolation} from "../model/RuleViolation";
+
+export default function RuleViolationDetails() {
+    const {id} = useParams<{ id: string}>();
+    const {isLoading, isError, violation} = useRuleViolation(id!);
+    const palette = useTheme().palette;
+
+    function RuleViolationInfoBar({violation} : {violation: RuleViolation}) {
```

Comment:
Weet niet of het beter is om functions uit main function te extracten (omdat ze groot worden en moeilijker te lezen wordt wat nu de effectieve return van de function is)?